internal/tools/terraformlinter/terraform_linter.go

```diff
@@ -0,0 +1,335 @@
+					depth = depth + 1
+				}
+				if t.Type == hclsyntax.TokenCBrace || t.Type == hclsyntax.TokenCBrack {
+					depth = depth - 1
+				}
+				if depth == 0 && t.Type == hclsyntax.TokenNewline {
+					// Check for an extra newline
+					trailingNewline := false
+					if len(tokens) > 0 && tokens[0].Type == hclsyntax.TokenNewline {
+						trailingNewline = true
+					}
+					position, ok := positionMap[contents]
+					if !ok {
+						position = Ignored
+					}
+					attrs = append(attrs, tokenAttr{tokenPos: position, token: token, trailingNewline: trailingNewline})
+					skipping = false
+				}
+			}
+		}
+	}
+	return generateViolations(attrs)
+}
+
+func generateViolations(idents []tokenAttr) []*ViolationInstance {
+	var instances []*ViolationInstance
+	var lastAttr tokenAttr
+	for pos, token := range idents {
+		contents := string(token.token.Bytes)
+		switch contents {
+		// for_each and count should be at the top
+		case attrForEach:
+			fallthrough
+		case attrCount:
+			if pos != 0 {
+				instances = append(instances, &ViolationInstance{ViolationType: fmt.Sprintf(errorLeadingMetaBlockAttribute, contents), Path: token.token.Range.Filename, Line: token.token.Range.Start.Line})
+			}
+		// provider is at the top but below for_each or count if they exist
+		case attrProvider:
+			if pos > 0 && lastAttr.tokenPos != Leading {
+				instances = append(instances, &ViolationInstance{ViolationType: fmt.Sprintf(errorLeadingMetaBlockAttribute, attrProvider), Path: token.token.Range.Filename, Line: token.token.Range.Start.Line})
+			}
+		// source should be the top attribute in a module
+		case attrSource:
+			if pos != 0 {
+				instances = append(instances, &ViolationInstance{ViolationType: fmt.Sprintf(errorLeadingMetaBlockAttribute, attrSource), Path: token.token.Range.Filename, Line: token.token.Range.Start.Line})
+			}
+		case attrDependsOn:
+			// depends_on somewhere above where it should be
+			if pos < len(idents)-1 && idents[len(idents)-1].tokenPos != Trailing {
+				instances = append(instances, &ViolationInstance{ViolationType: fmt.Sprintf(errorTrailingMetaBlockAttribute, attrDependsOn), Path: token.token.Range.Filename, Line: token.token.Range.Start.Line})
+			}
+			// depends_on after lifecycle
+			if pos == len(idents)-1 && lastAttr.tokenPos == Trailing {
+				instances = append(instances, &ViolationInstance{ViolationType: fmt.Sprintf(errorTrailingMetaBlockAttribute, attrDependsOn), Path: token.token.Range.Filename, Line: token.token.Range.Start.Line})
+			}
+		case attrLifecycle:
+			// lifecycle should be last
+			if pos != len(idents)-1 {
+				instances = append(instances, &ViolationInstance{ViolationType: fmt.Sprintf(errorTrailingMetaBlockAttribute, attrLifecycle), Path: token.token.Range.Filename, Line: token.token.Range.Start.Line})
+			}
+		// All provider specific entries follow the same logic. Should be below the metadata segment and above everything else
+		case attrProviderProject:
+			fallthrough
```

Comment:
I think this is the first time I've seen `fallthrough` used. Why do we need it?

Reply:
Mainly because I hate repeating myself but I'm going to have to change this to force ordering on these attributes anyhow so there won't be any opportunity to fallthrough after that change.